File: ProyectoenJava/MainActivity.java
package com.example.proyectojavalab; // <-- REVISA QUE ESTO COINCIDA CON TU PROYECTO

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    // Declaración de objetos (En Java se hace fuera del onCreate)
    private EditText etNombre;
    private Button btnSaludar;
    private TextView tvResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Inicialización de componentes
        etNombre = findViewById(R.id.etNombreJava);
        btnSaludar = findViewById(R.id.btnSaludarJava);
        tvResultado = findViewById(R.id.tvResultadoJava);

        // Evento Clic al estilo Java (Clase anónima)
        btnSaludar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String nombre = etNombre.getText().toString().trim();

                if (nombre.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Por favor, escribe un nombre", Toast.LENGTH_SHORT).show();
                } else {
                    String mensaje = "¡Hola " + nombre + ", bienvenido a Java!";
                    tvResultado.setText(mensaje);
                    Toast.makeText(MainActivity.this, "Procesado correctamente", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: ActvidadMoneda/MainActivity.kt
package com.example.moneda

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.RadioButton
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    // Declaración de variables para los componentes (Fase 4)
    private lateinit var cantidadInput: EditText
    private lateinit var radioSoles: RadioButton
    private lateinit var radioDolares: RadioButton

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Inicialización de componentes
        cantidadInput = findViewById(R.id.editTextText)
        radioDolares = findViewById(R.id.radio0)
        radioSoles = findViewById(R.id.radio1)
    }

    // El nombre de esta función debe ser igual al que pusiste en el XML: android:onClick="miClicManejador"
    fun miClicManejador(view: View) {
        val textoCantidad = cantidadInput.text.toString()

        if (textoCantidad.isEmpty()) {
            Toast.makeText(this, "Por favor, ingresa una cantidad", Toast.LENGTH_SHORT).show()
            return
        }

        val cantidad = textoCantidad.toDouble()
        var resultado = 0.0

        // Lógica de conversión (Usando un tipo de cambio de ejemplo)
        if (radioDolares.isChecked) {
            // "A soles": Supongamos que 1 dólar = 3.80 soles
            resultado = cantidad * 3.80
            Toast.makeText(this, "El resultado en soles es: S/ $resultado", Toast.LENGTH_LONG).show()
        } else if (radioSoles.isChecked) {
            // "A dólares": Supongamos que 1 sol = 0.26 dólares
            resultado = cantidad / 3.80
            Toast.makeText(this, "El resultado en dólares es: $ $resultado", Toast.LENGTH_LONG).show()
        }
    }
}